type_classes.md

```diff
@@ -298,6 +542,10 @@ def getUnit [Monoid α] : α :=
   1
 ```
 
+#check getUnit fails???  Why can't I call it, it has only implicit args right?  What
```

Comment:
thanks.

Reply:
Thanks, how does the body of getUnit with the simple value `1` play into all this?  It seems to be ignored and it returns  the MulMonoid  unit, for example:

```lean
instance MulMonoid : Monoid Nat where
  unit := 5
  op := Nat.mul

set_option pp.all true
#check (getUnit : Nat)  -- @getUnit.{0} Nat MulMonoid : Nat
#eval (getUnit : Nat)   -- 5
#eval 
```

is there some other way to write the getUnit function to make it a bit more clear what it is doing?

Reply:
Yes that would certainly be more clear....the `1`here is coming from this instance:
```lean
instance [s : Monoid α] : OfNat α (nat_lit 1) where
  ofNat := s.unit
```
It's basically telling to lean that it can turn a `1` into something of type `α` by replacing it with the value behind `ofNat` (`s.unit`) in this case. This is done here since mathematicians have the tendency to denote the neutral element of multiplicative Monoids as `1`. But the notation is probably doing more harm than good here so removing that part would be totally fine I think.